test/CustomOperatorTest/TestOperator.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace CustomOperatorTest\Rx\Operator;
+
+use Rx\ObservableInterface;
+use Rx\ObserverInterface;
+use Rx\Operator\OperatorInterface;
+use Rx\SchedulerInterface;
+
+class TestOperator implements OperatorInterface
+{
+    private $mapTo;
+
+    public function __construct($mapTo)
+    {
+        $this->mapTo = $mapTo;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function __invoke(
+        ObservableInterface $observable,
+        ObserverInterface $observer,
+        SchedulerInterface $scheduler = null
+    ) {
+        return $observable
```

Comment:
You are right - but this feels like it should be its own issue as it is done everywhere - we will end up feeding everything through `->asObservable`. I believe there was a suggestion to drop the ObservableInterface before but I cannot find it.

This cannot be changed to `Observable` though because the `OperatorInterface` does not match.

I will open another issue for this discussion.